pkgs/build-support/replace-secret/replace-secret.py

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env python
+
+import click
+
+@click.command()
+@click.argument('string_to_replace')
+@click.argument('secret_filename')
+@click.argument('filename')
+def replace(string_to_replace, secret_filename, filename):
+    """Replace a string in one file with a secret from a second file.
+
+    Since the secret is read from a file, it won't be leaked through
+    '/proc/<pid>/cmdline', unlike when 'sed' or 'replace' is used.
+
+    """
+    with open(secret_filename) as sf, open(filename, 'r+') as f:
```

Comment:
Should this set a umask or something?

Reply:
It does the replacement in-place, so the file's permissions retained. I guess it could force the permissions to something restrictive, but this feels like a surprising behavior..